### DISCUSSION TITLE
What's the best method for draping a single WGS84 texture over the globe?

### DISCUSSION THREAD

**Discussion by robhawkes** [iTowns/itowns#1708] [Q&A]
Hello,

I'm hoping to use iTowns again for another project of mine, but I'm stuck working out the best method for draping a single texture over the entire globe (eg. a single WGS84 image that covers the extent -180, 180, -90, 90).

For example (though this isn't actually the full extend as it's a cropped screenshot):

![image](https://user-images.githubusercontent.com/22612/130366199-1e51cf13-949f-4993-a839-3294661b3b26.png)

It seems there isn't a layer for this in the docs, and I've had limited success forcefully updating the material set on the globe geometry, which seems like more of a hack anyway so I'd like to avoid that.

Is there a layer that I'm missing that could do this, or will I need to dig into the guts of the framework to add my own sphere geometry over the top of the terrain?

Thanks!

**Comment by gchoqueux** [comment]
in the case of a WGS84 projection in `globeView`, there's no simple method because tile matrix is divided by 2.
You could divide, on Y axe,  your image in 2 images, with name : `'wgs84_0.png` and `wgs84_1.png`.
 
and use this code

```js
var source_wgs84 = new itowns.TMSSource({
    crs: 'EPSG:4326',
    url: 'wgs84_${x}.png',
    zoom: { min:0, max: 0}
});
var layer_wgs84 = new itowns.ColorLayer('wgs84', {
    source: source_wgs84
});
view.addLayer(layer_wgs84);
```

![wgs84_0](https://user-images.githubusercontent.com/11291849/130630928-6e0d5da4-7428-46a7-999b-b7a96e2024d7.png)
![wgs84_1](https://user-images.githubusercontent.com/11291849/130630938-ec0b84b0-eb6b-4013-b10b-1e8f1c4ac1e5.png)

![image](https://user-images.githubusercontent.com/11291849/130631404-95bb1410-28a2-4794-97b4-054b7ea584da.png)
